File: sample/src/main/java/contacts/sample/cheatsheet/sim/java/DeleteSimContactsActivity.java
package contacts.sample.cheatsheet.sim.java;

import android.app.Activity;

import contacts.core.ContactsFactory;
import contacts.core.entities.SimContact;
import contacts.core.sim.SimContactsDelete;

public class DeleteSimContactsActivity extends Activity {

    SimContactsDelete.Result deleteSimContact(SimContact simContact) {
        return ContactsFactory.create(this).sim().delete().simContacts(simContact).commit();
    }

    SimContactsDelete.Result deleteSimContactWithNameAndNumber(String name, String number) {
        return ContactsFactory.create(this).sim().delete().simContact(name, number).commit();
    }
}

File: sample/src/main/java/contacts/sample/cheatsheet/sim/kotlin/DeleteSimContactsActivity.kt
package contacts.sample.cheatsheet.sim.kotlin

import android.app.Activity
import contacts.core.Contacts
import contacts.core.entities.SimContact
import contacts.core.sim.SimContactsDelete

class DeleteSimContactsActivity : Activity() {

    fun deleteSimContact(simContact: SimContact): SimContactsDelete.Result =
        Contacts(this).sim().delete().simContacts(simContact).commit()

    fun deleteSimContactWithNameAndNumber(name: String?, number: String?): SimContactsDelete.Result =
        Contacts(this).sim().delete().simContact(name, number).commit()
}